python/ray/llm/_internal/serve/serving_patterns/data_parallel/dp_group_manager.py

```diff
@@ -0,0 +1,313 @@
+import asyncio
+import logging
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Tuple
+
+import ray
+from ray.serve.schema import ReplicaRank
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class GroupInfo:
+    """State for a single DP group.
+
+    Attributes:
+        master_info: Tuple of (dp_address, dp_rpc_port) for the group's master.
+        dp_rank_to_replica_id: Mapping from DP rank to replica ID for tracking
+            registered replicas.
+        master_info_event: Event that is set when master info is available.
+    """
+
+    master_info: Optional[Tuple[str, int]] = None
+    dp_rank_to_replica_id: Dict[int, str] = field(default_factory=dict)
+    master_info_event: asyncio.Event = field(default_factory=asyncio.Event)
+
+
+class DPGroupManager:
+    """Data Parallel Group Manager.
+
+    This actor manages multiple DP groups within a single deployment. Each group
+    is a set of replicas that participate in the same collective operations.
+
+    Key concepts:
+    - dp_group_size: Number of replicas in each DP group (e.g., 8)
+    - dp_size_per_node: Number of DP replicas that fit on a node, determined by
```

Comment:
I don't think this description is accurate:

determined by node count / tp size ? 
but then what is the difference beteween this and num_replica_per_node ?